jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/PdfHeader.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * 
+ */
+package edu.harvard.hul.ois.jhove.module.pdf;
+
+import java.io.IOException;
+
+/**
+ * Simple class that is the a prototype of a proper header parser class. The aim
+ * was to introduce a simple version check for the PDF/A minor version number,
+ * see {@link PdfHeader#isVersionValid()}, while not changing anything else
+ * through over ambition.
+ * 
+ * @author <a href="mailto:carl@openpreservation.org">Carl Wilson</a>
+ *         <a href="https://github.com/carlwilson">carlwilson AT github</a>
+ * @version 0.1 Created 8 Mar 2018:00:46:39
+ */
+
+public final class PdfHeader {
+	public static final String PDF_VER1_HEADER_PREFIX = "PDF-1."; //$NON-NLS-1$
+	public static final String PDF_SIG_HEADER = "%" + PDF_VER1_HEADER_PREFIX; //$NON-NLS-1$
+	public static final String POSTSCRIPT_HEADER_PREFIX = "!PS-Adobe-"; //$NON-NLS-1$
+	public static final int MAX_VALID_MAJOR_VERSION = 7;
+
+	private final String versionString;
+	private final boolean isPdfACompilant;
+
+	/**
+	 * 
+	 */
+	private PdfHeader(final String versionString,
+			final boolean isPdfaCompliant) {
+		this.versionString = versionString;
+		this.isPdfACompilant = isPdfaCompliant;
+	}
+
+	/**
+	 * @return the version string parsed from the PDF Header
+	 */
+	public String getVersionString() {
+		return this.versionString;
+	}
+
+	/**
+	 * @return true if the header is considered PDF/A compliant, otherwise false
+	 */
+	public boolean isPdfACompliant() {
+		return this.isPdfACompilant;
+	}
+
+	/**
+	 * Performs a very simple version number validity check. Given version
+	 * number is a String of form 1.x, x is the minor version number. This
+	 * method parses the minor version number from the version String and tests
+	 * whether it is less than or equal to
+	 * {@link PdfHeader#MAX_VALID_MAJOR_VERSION}.
+	 * 
+	 * @return true if an integer minor version number can be parsed from the
+	 *         version string AND it is less than or equal to
+	 *         {@link PdfHeader#MAX_VALID_MAJOR_VERSION}. Otherwise false.
+	 */
+	public boolean isVersionValid() {
+		// Set minor version to one larger than maximum so invalid if parse
+		// fails
+		int minorVersion = MAX_VALID_MAJOR_VERSION + 1;
+		try {
+			minorVersion = getMinorVersion(this.versionString);
+		} catch (NumberFormatException nfe) {
+			// TODO : Do nothing for now, the version number is still invalid.
```

Comment:
TODO: Nothing?

Reply:
The aim is to not break existing behaviour. This ensures that this case is handled as before as the minor version number is set too large and false is returned. I've left the TODO as I'm intending to polish this a little more, but it needs a new error messages and is outside of the PR scope.

Reply:
I wonder if in the long term view it is as easy as thinking about the minor version not being set too large, especially with PDF 2.0 now available. Couldn't it be instead checking the versionString against a list of currently available versions? So, as of today: 1.0, 1.1, 1.2, 1.3, ,1.4, 1.5., 1.6, 1.7, 2.0 ? 
Downside is that the list of available versions would have to be updated everytime a new version becomes available - but if the code instead states that the max correct minor version is 7, which is true for PDF 1.x, it would have to still be updated regarding the limitation of minor version for 2.x as there currently only is 2.0 (and this would again have to be updated once 2.1 becomes available).
Does that make any sense?